### PR TITLE
[FIX] Comunity users accessing exclusive ee endpoints in omnichannel section

### PR DESCRIPTION
**Pull request by FabioCavaleti**
Solving task OC-587

Fix some message errors that were showing in the community edition because some improper endpoints were being accessed.
I fix the error when clicking on close room icon, fix the unauthorized access to units page, and change some deprecated endpointData to useQuery function to calls an endpoint.

### CODE REVIEW

File: apps/meteor/ee/client/omnichannel/tags/TagEditWithData.js [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
-import { AsyncStatePhase } from '../../../../client/hooks/useAsyncState';
-import { useEndpointData } from '../../../../client/hooks/useEndpointData';
 import TagEdit from './TagEdit';
 import TagEditWithDepartmentData from './TagEditWithDepartmentData';
 
 function TagEditWithData({ tagId, reload, title }) {
-	const { value: data, phase: state, error } = useEndpointData('/v1/livechat/tags/:tagId', { keys: { tagId } });
-
+	const getTag = useEndpoint('GET', '/v1/livechat/tags/:tagId', { tagId });
+	const { data, isLoading, isError } = useQuery(['/v1/livechat/tags/:tagId', tagId], () => getTag(), { enabled: Boolean(tagId) });
```
Comment (KevLehman) [inline apps/meteor/ee/client/omnichannel/tags/TagEditWithData.js:12]: Why in the other one you had `!!value` and here `Boolean(value)`? Should we keep consistency?

File: apps/meteor/ee/client/omnichannel/units/UnitsRoute.js [hunk 2]
```diff
@@ -60,7 +62,9 @@ function UnitsRoute() {
 			}),
 	);
 
-	const { value: data = {}, reload } = useEndpointData('/v1/livechat/units', { params: query });
+	// const { value: data = {}, reload } = useEndpointData('/v1/livechat/units', { params: query });
```
Comment (KevLehman) [inline apps/meteor/ee/client/omnichannel/units/UnitsRoute.js:65]: If not needed, remove :)

File: apps/meteor/ee/client/omnichannel/tags/TagEditWithData.js (hunk 1 above)
Comment (FabioCavaleti) [reply apps/meteor/ee/client/omnichannel/tags/TagEditWithData.js:12]: You are completely right!! I was doing some tests and I ended up forgetting to make things consistent. Thanks, Kevin

**Review by KevLehman — DISMISSED** [review]

File: apps/meteor/client/components/Omnichannel/Tags.tsx [hunk 3]
```diff
@@ -23,13 +22,18 @@ const Tags = ({
...
 	// TODO: Refactor the formsSubscription to use components instead of hooks (since the only thing the hook does is return a component)
 	const { useCurrentChatTags } = forms;
 	// Conditional hook was required since the whole formSubscription uses hooks in an incorrect manner
 	const EETagsComponent = useCurrentChatTags?.();
 
+	const getTags = useEndpoint('GET', '/v1/livechat/tags');
+	const {
+		data: tagsResult,
+		isLoading,
+		isFetching,
```
Comment (aleksandernsilva) [inline apps/meteor/client/components/Omnichannel/Tags.tsx:34]: You can use here `isInitialLoading` which is equivalent to `isFetching && isLoading`

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]

**Review by aleksandernsilva — APPROVED** [review]

File: apps/meteor/ee/client/omnichannel/units/UnitsRoute.js [hunk 4]
```diff
@@ -31,13 +32,14 @@ const useQuery = ({ text, itemsPerPage, current }, [column, direction]) =>
 function UnitsRoute() {
 	const t = useTranslation();
 	const canViewUnits = usePermission('manage-livechat-units');
+	const license = useHasLicenseModule('livechat-enterprise');
```
Comment (aleksandernsilva) [inline apps/meteor/ee/client/omnichannel/units/UnitsRoute.js:35]: I recommend naming this const as `isEnterprise`

File: apps/meteor/ee/client/omnichannel/units/UnitsRoute.js [hunk 5]
```diff
@@ -103,7 +106,7 @@ function UnitsRoute() {
 		return <UnitEdit title={t('New_Unit')} reload={reload} isNew={true} />;
 	}
 
-	if (!canViewUnits) {
+	if ((license && !canViewUnits) || !license) {
```
Comment (aleksandernsilva) [inline apps/meteor/ee/client/omnichannel/units/UnitsRoute.js:109]: I belive this is a better way to do this check `!license || !canViewUnits`

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]

**Pull request merged**